test/service_tests/country_institute_api_test.dart

```diff
@@ -0,0 +1,103 @@
+import 'dart:convert';
+
+import 'package:flutter_test/flutter_test.dart';
+import 'package:http/http.dart';
+import 'package:http/testing.dart';
+import 'package:mobile_app/models/failure_model.dart';
+import 'package:mobile_app/utils/api_utils.dart';
+import 'package:mobile_app/services/API/country_institute_api.dart';
+
+import '../setup/test_data/mock_country_data.dart';
+import '../setup/test_data/mock_institutes_data.dart';
+
+void main() {
+  group('CountryApi -', () {
+    group('fetchCountries -', () {
+      test('When called & http client returns succes response', () async {
+        ApiUtils.client = MockClient(
+            (_) => Future.value(Response(jsonEncode(mockCountries), 200)));
+        var _countryApi = HttpCountryInstituteAPI();
+
+        expect(
```

Comment:
Please reformat the expect blocks, hard to comprehend. Can be done in a line or two. No need to add redundant commas.